WebDriverAgentLib/Utilities/FBScreenshot.m

```diff
@@ -235,6 +158,26 @@ + (nullable id)imageEncodingWithUniformTypeIdentifier:(NSString *)uti
      buildError:error];
     return nil;
   }
+
+  if ([uti isEqualToString:UTTypeHEIC.identifier]) {
+    static BOOL isHeicSuppported = NO;
+    static dispatch_once_t onceToken;
+    dispatch_once(&onceToken, ^{
+      SEL selector = NSSelectorFromString(@"supportsHEICImageEncoding");
+      NSMethodSignature *signature = [imageEncodingClass methodSignatureForSelector:selector];
+      if (nil != signature) {
+        NSInvocation *invocation = [NSInvocation invocationWithMethodSignature:signature];
+        [invocation setSelector:selector];
+        [invocation invokeWithTarget:imageEncodingClass];
+        [invocation getReturnValue:&isHeicSuppported];
+      }
+    });
+    if (!isHeicSuppported) {
+      [FBLogger logFmt:@"The device under test does not support HEIC image encoding. Falling back to PNG"];
```

Comment:
@KazuCocoa Could you please check how this validation behaves on real devices?

Reply:
Yes, of course. Potentially it will be late a bit, but will do

Reply:
Maybe I can do this tonight in PT